src/scheduler/work_bucket.rs

```diff
@@ -43,6 +43,8 @@ impl<VM: VMBinding> BucketQueue<VM> {
     }
 }
 
+// Allow type_complexity as Rust support for trait aliases is still experimental
+#[allow(clippy::type_complexity)]
```

Comment:
I think it is complaining about `can_open: Option<Box<dyn (Fn(&GCWorkScheduler<VM>) -> bool) + Send>>`, which is indeed too complex. We can create a type alias:
```rust
pub type BucketOpenCondiiton<VM> = Box<dyn (Fn(&GCWorkScheduler<VM>) -> bool) + Send>;
```
This should make the warning disappear.

Reply:
I tried doing that but then Rust says "type aliases can't be used as traits"

Reply:
Ah I see. I made a new trait called `trait BucketOpenCondiiton<VM: VMBinding> Fn(&GCWorkScheduler<VM>) -> bool) + Send` and Rust was complaining about that. I'll use your fix :+1:

Reply:
In the future we may want to make this a trait alias instead of a type alias so that the `set_open_condition` function can also use the trait directly.